### PR TITLE
fix(auth): enable non-https OpenID Connect config URLs for development with local IDPs

### LINKED ISSUE #5417: [BUG]OAuth2 Disable https-check for localhost domain
**Describe the bug**
When I setup a playground with a custom IDP i.e. Keycloak. It runs under localhost without any ssl-certificate which is not necessary for local development.

**Expected behavior**
The https-check should not be done for localhost urls.

**Errors from docker-compose**
hoenix-1  | ✅ Migrations complete.
phoenix-1  | Traceback (most recent call last):
phoenix-1  |   File "<frozen runpy>", line 198, in _run_module_as_main
phoenix-1  |   File "<frozen runpy>", line 88, in _run_code
phoenix-1  |   File "/phoenix/env/phoenix/server/main.py", line 437, in <module>
phoenix-1  |     main()
phoenix-1  |   File "/phoenix/env/phoenix/server/main.py", line 420, in main
phoenix-1  |     oauth2_client_configs=get_env_oauth2_settings(),
phoenix-1  |                           ^^^^^^^^^^^^^^^^^^^^^^^^^
phoenix-1  |   File "/phoenix/env/phoenix/config.py", line 456, in get_env_oauth2_settings
phoenix-1  |     return [OAuth2ClientConfig.from_env(idp_name) for idp_name in sorted(idp_names)]
phoenix-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
phoenix-1  |   File "/phoenix/env/phoenix/config.py", line 456, in <listcomp>
phoenix-1  |     return [OAuth2ClientConfig.from_env(idp_name) for idp_name in sorted(idp_names)]
phoenix-1  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
phoenix-1  |   File "/phoenix/env/phoenix/config.py", line 428, in from_env
phoenix-1  |     raise ValueError(
phoenix-1  | ValueError: Server metadata URL for auth0 OAuth2 IDP must be a valid URL using the https protocol

**Environment (please complete the following information):**
 - OS: Mac OS 14.6.1, M2
 - Notebook Runtime VSCode
 - Browser Chrome 130.0.6723.117 for arm64

### PR DESCRIPTION
**Pull request by sebibrunner**
When using open-source identity services such as Keycloak, users want to be able to run the identity service locally and point Phoenix toward a local OpenID Connect config URL during development.

resolves #5417

### CODE REVIEW

**Comment by sebibrunner** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/phoenix/config.py [hunk 1]
```diff
@@ -424,9 +424,9 @@ def from_env(cls, idp_name: str) -> "OAuth2ClientConfig":
                 f"An OpenID Connect configuration URL must be set for the {idp_name} OAuth2 IDP "
                 f"via the {oidc_config_url_env_var} environment variable"
             )
-        if urlparse(oidc_config_url).scheme != "https":
+        if urlparse(oidc_config_url).scheme != "https" and urlparse(oidc_config_url).hostname != "localhost":
```
Comment (axiomofjoy) [inline src/phoenix/config.py:427]:
```suggestion
        parsed_oidc_config_url = urlparse(oidc_config_url)
        is_local_oidc_config_url = parsed_oidc_config_url.hostname in ("localhost", "127.0.0.1")
        if parsed_oidc_config_url.scheme != "https" and not is_local_oidc_config_url:
```

Let's check for both `127.0.0.1` and `localhost`

File: src/phoenix/config.py [hunk 2]
```diff
@@ -424,9 +424,9 @@ def from_env(cls, idp_name: str) -> "OAuth2ClientConfig":
...
             raise ValueError(
-                f"Server metadata URL for {idp_name} OAuth2 IDP "
+                f"Server metadata URL for {idp_name} OAuth2 IDP which is not localhost"
```
Comment (axiomofjoy) [inline src/phoenix/config.py:429]: I think the existing error message is probably fine as is 😄

**Review by axiomofjoy — COMMENTED** [review]
Thanks!

File: src/phoenix/config.py (hunk 1 above)
Comment (sebibrunner) [reply src/phoenix/config.py]: Thank you for the review. I also thaught about that, but then somehow didn't decided to add it.

**Comment by axiomofjoy** [comment]
thanks @sebibrunner! this will be included in the next version of phoenix

**Pull request merged**